src/chat/slashCommands.ts

```diff
@@ -145,8 +151,17 @@ export class SlashCommandsOwner implements IAgentRequestHandler {
         }
     }
 
-    public getSlashCommands(): ([string, SlashCommandConfig])[] {
-        return Array.from(this._invokeableSlashCommands.entries());
+    public async getSlashCommands(): Promise<SlashCommands> {
+        await this._callLazyInvokeableSlashCommandsResolvers();
+        return new Map(Array.from(this._invokeableSlashCommands.entries()));
+    }
+
+    private async _callLazyInvokeableSlashCommandsResolvers(): Promise<void> {
+        for (const resolver of this._invokeableSlashCommandsResolvers) {
+            const lazySlashCommands = await resolver();
+            this.addInvokeableSlashCommands(lazySlashCommands);
+        }
```

Comment:
This eventually awaits the extensions' activation, correct? I think you can "parallelize" by doing something like this:

```suggestion
        const invokeableSlashCommandResolverPromises: Promise<void> = [];
        for (const resolver of this._invokeableSlashCommandsResolvers) {
            const promise = resolver().then(resolvedSlashCommands => this.addInvokeableSlashCommands(resolvedSlashCommands), rejectErr => { throw rejectErr; /* Or whatever */});
            invokeableSlashCommandResolverPromises.push(promise);
        }
        await Promise.all(...invokeableSlashCommandResolverPromises);
```

Reply:
Sure.